internal/evaluator/evaluator.go

```diff
@@ -98,10 +101,30 @@ func (r *ConfigurationEvaluatorServiceServer) EvaluateConfig(
 		yamlConfigs = append(yamlConfigs, generatedYamlConfig)
 	}
 
+	additionalInstances := make(map[string]protoreflect.MessageDescriptor)
```

Comment:
It would be nice if this added piece of code would provide an explanation of what it's doing and some test, since it's somewhat complex and isn't covered at all.

Reply:
Added a test on the CLI side in 9795392c9c1b4abcdd2d8a4d3e8d66dd206a70a4